apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/commons/AbstractSamhsaMatcherTest.java

```diff
@@ -0,0 +1,812 @@
+package gov.cms.bfd.server.war.commons;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anySet;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import gov.cms.bfd.model.codebook.data.CcwCodebookVariable;
+import gov.cms.bfd.server.war.adapters.CodeableConcept;
+import gov.cms.bfd.server.war.adapters.Coding;
+import gov.cms.bfd.server.war.adapters.DiagnosisComponent;
+import gov.cms.bfd.server.war.adapters.FhirResource;
+import gov.cms.bfd.server.war.adapters.ItemComponent;
+import gov.cms.bfd.server.war.adapters.ProcedureComponent;
+import gov.cms.bfd.server.war.utils.ReflectionTestUtils;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.ArgumentCaptor;
+
+@RunWith(Enclosed.class)
+public class AbstractSamhsaMatcherTest {
+
+  private static final IllegalStateException INVOCATION_EXCEPTION =
+      new IllegalStateException("Method was not called with expected arguments for test");
+
+  @FunctionalInterface
+  public interface SamhsaFilterMethod<T> {
+    boolean apply(AbstractSamhsaMatcher<IBaseResource> matcher, T coding);
+  }
+
+  public static class NonParameterizedTests {
+
+    @Test
+    public void shouldReturnColumnValues() {
+      List<String> expected = List.of("1", "2", "3", "4");
+      List<String> actual =
+          AbstractSamhsaMatcher.resourceCsvColumnToList("samhsa_codes_test_file.csv", "columnB");
+
+      assertEquals(expected, actual);
+    }
+
+    @Test
```

Comment:
I typically just document the scenario in the test name, in this one's case: `shouldReturnTrueForSamhsaProcedureComponent`.  Would we want to add a java doc that states the same thing or is there additional information that would help here?

Reply:
When I document tests I typically use the additional space to more clearly define the test parameters, so the intent of the test is clearly described. This can sometimes mean explaining a bit of context beyond the test name that might not be evident to someone seeing it for the first time.

For instance, for this test, I might write something like this:
```
/**
  * Verifies that when the SamhsaMatcher calls containsSamhsaIcdProcedureCode on 
  * a list of multiple procedures, where at least one is determined to be a SAMHSA ICD procedure 
  * and some are not, the result is {@code true}. If ANY procedure codes are determined to be a match,
  * the matcher must return true.
  *
**/
```

I would consider this the formal writeup of the test case; describe what, how, and why this is being tested, along with the result, as well as any required context.